File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Greece.m
//
//  UIImage+Greece.m
//  CountriesFlags
//

#import "Countries/UIImage+Greece.h"

@implementation UIImage (Greece)

+ (UIImage *)flagGreeceWithSize:(CGSize)size {
    UIColor* colorWhite = [UIColor whiteColor];
    UIColor* colorBlue = [UIColor colorWithRed:13.0f/255.0f
                                         green:94.0f/255.0f
                                          blue:175.0f/255.0f
                                         alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    for (int i = 2; i <= 24; i += 4) {
        UIRectFill(CGRectMake(0.0f, size.height / 18.0f * i, size.width, size.height / 18.0f * 2.0f));
    }
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width / 27.0f * 10.0f, size.height / 18.0f * 10.0f));
    
    [colorWhite setFill];
    // horizontal line
    UIRectFill(CGRectMake(0.0f, size.height / 18.0f * 4.0f, size.width / 27.0f * 10.0f, size.height / 18.0f * 2.0f));
    // vertical line - add 0.5 to remove blue line for small images
    UIRectFill(CGRectMake(size.width / 27.0f * 4.0f, 0.0f, size.width / 27.0f * 2.0f, size.height / 18.0f * 10.5f));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image != nil ? image : [UIImage new];
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Greece.swift
//
//  UIImage+Greece.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Greece "wikipedia"
     
     Generates the flag of Greece.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Greece.
     */
    internal static func flagGreece(with size: CGSize) -> UIImage? {
        let colorBlue = UIColor(red: 13.0 / 255.0,
                                green: 94.0 / 255.0,
                                blue: 175.0 / 255.0,
                                alpha: 1.0)
        let colorWhite = UIColor.white

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorBlue.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        for index in stride(from: 2, through: 14, by: 4) {
            UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 18.0 * CGFloat(index)),
                              size: CGSize(width: size.width, height: size.height / 18.0 * 2.0)))
        }

        colorBlue.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: 0.0),
                          size: CGSize(width: size.width / 27.0 * 10.0, height: size.height / 18.0 * 10.0)))

        colorWhite.setFill()
        // horizontal line
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 18.0 * 4.0),
                          size: CGSize(width: size.width / 27.0 * 10.0, height: size.height / 18.0 * 2.0)))
        // vertical line - add 0.5 to remove blue line for small images
        UIRectFill(CGRect(origin: CGPoint(x: size.width / 27.0 * 4.0, y: 0.0),
                          size: CGSize(width: size.width / 27.0 * 2.0, height: size.height / 18.0 * 10.5)))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
